varnish-sys/src/vcl/ws_str_buffer.rs

```diff
@@ -0,0 +1,391 @@
+use crate::ffi;
+use crate::ffi::VCL_STRING;
+#[cfg(not(varnishsys_6))]
+use crate::ffi::{vrt_blob, VCL_BLOB};
+use crate::vcl::VclError::WsOutOfMemory;
+use crate::vcl::VclResult;
+use std::cell::RefCell;
+use std::io::Write;
+use std::marker::PhantomData;
+use std::num::NonZeroUsize;
+use std::slice::from_raw_parts_mut;
+use std::{io, mem, ptr};
+
+/// The free region of the workspace that functions as a "resizable" vector, up to the end of the workspace.
+/// The buffer must be finalized using `finish()` to avoid being reclaimed when dropped.
+#[derive(Debug)]
+pub struct WsBuffer<'a, Prefix, Item, Suffix, Output> {
+    /// The workspace pointer, used to release the workspace
+    ws: &'a mut ffi::ws,
+    /// The start of the writable buffer, aligned to the content type. Will set to null when finished.
+    start: *mut Item,
+    /// The reserved buffer will move its start as we write to it
+    buffer: &'a mut [Item],
+
+    _prefix: PhantomData<Prefix>,
+    _output: PhantomData<Output>,
+    _suffix: PhantomData<Suffix>,
+}
```

Comment:
it's quite a big dance just to allow `VCL_BLOB`, I'd much rather we had a `new_blob(&Ctx, &[u8])` constructor.

If you really have to, then you can add a `finalize_as_blob()` to the byte buffer, and stash `vrt_blob` at the end of the buffer, but this whole machinery makes me uneasy